### PR TITLE
ref bug#133805 set time by branch

### PR DESCRIPTION
**Pull request by HyperionsThs**
Link : https://dev.framgia.com/redmine/issues/133805
Effect : N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java [hunk 1]
```diff
@@ -3,37 +3,55 @@
...
+import java.text.ParseException;
+import java.util.List;
 
 /**
  * Created by ASUS on 13/06/2017.
  */
 
 public class ItemListRequestViewModel extends BaseObservable {
 
+    private final String TAG = "ItemListRequestViewModel";
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java]: WARNING: Name 'TAG' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java [hunk 2]
```diff
@@ -42,59 +60,100 @@
...
+                    case LeaveType.LEAVE_EARLY_A:
+                    case LeaveType.LEAVE_EARLY_WOMAN_M:
+                    case LeaveType.LEAVE_EARLY_WOMAN_A:
+                    case LeaveType.FORGOT_TO_CHECK_OUT:
+                        return requestCheckOutByTimeAndDate;
+                    default:
+                        return "";
+                }
+            } catch (ParseException exception) {
+                Log.e(TAG, exception.getLocalizedMessage());
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java]: ERROR: [LongLogTag] The logging tag can be at most 23 characters, was 24 (ItemListRequestViewModel)
 Log tags are only allowed to be at most 23 tag characters long.

File: app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java [hunk 3]
```diff
@@ -3,37 +3,56 @@
...
     ItemListRequestViewModel(Object object,
-            BaseRecyclerViewAdapter.OnRecyclerViewItemClickListener<Object> itemClickListener) {
+            BaseRecyclerViewAdapter.OnRecyclerViewItemClickListener<Object> itemClickListener,
+            User user) {
         mObject = object;
         mItemClickListener = itemClickListener;
         if (object instanceof LeaveRequest) {
             mRequest = (LeaveRequest) object;
+            if (user.getSpecial() == UserType.CHILDREN) {
+                mExtraTime = 15;
```
Comment (tranminhtri911) [inline app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java]: hard code

File: app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java [hunk 4]
```diff
@@ -42,59 +61,99 @@
...
+                        return requestCheckOutByTimeAndDate;
+                    case LeaveType.LEAVE_EARLY_WOMAN_A:
+                    case LeaveType.LEAVE_EARLY_A:
+                        return requestCheckOutByTime
+                                + Constant.BLANK_DASH_BLANK
+                                + timeOut
+                                + Constant.BLANK
+                                + requestCheckOutByDate;
+                    default:
+                        return "";
```
Comment (tranminhtri911) [inline app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java]: break;

File: app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java (hunk 3 above)
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java]: hard code

File: app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java [hunk 5]
```diff
@@ -3,37 +3,56 @@
...
+    private final String TAG = "ItemListRequestViewModel";
+
     private final Object mObject;
     private LeaveRequest mRequest;
     private OffRequest mRequestOff;
     private RequestOverTime mRequestOverTime;
     private int mStatusImage;
     private final BaseRecyclerViewAdapter.OnRecyclerViewItemClickListener<Object>
             mItemClickListener;
+    private int mExtraTime = 0;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java]: pls remove init value for this variable

File: app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java (hunk 3 above)
Comment (daolq3012) [reply app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java]: mExtraTime = user.getSpecial() == UserType.CHILDREN? 15: 0;

File: app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java [hunk 6]
```diff
@@ -42,59 +61,99 @@
...
+        for (int i = 0; i < branchList.size(); i++) {
+            if (branchList.get(i).getBranchId() == branchId) {
+                return i;
+            }
+        }
+        return 0;
+    }
+
+    private Shifts getShifts() {
+        return mUser.getBranches().get(positionBranchLeaveRequest).getShifts().get(0);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java]: pls add TODO in here!

File: app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java [hunk 7]
```diff
@@ -42,8 +56,35 @@
...
+    private int searchPositionBranch(List<Branch> branchList, int branchId) {
+        for (int i = 0; i < branchList.size(); i++) {
+            if (branchList.get(i).getBranchId() == branchId) {
+                return i;
+            }
+        }
+        return 0;
+    }
+
+    private Shifts getShifts() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/listrequest/ItemListRequestViewModel.java]: pls rename ``getShifts`` to ``getCurrentShift``

**Pull request merged**